File: s3-inf-ppo/laboratoria/lab07/php/src/ParkingLot.php
<?php

namespace ParkingLotManager;

use ParkingLotManager\Entities\Car;
use ParkingLotManager\Interfaces\EntityInterface;

final class ParkingLot {

	protected $entitiesOnProperty = [];
	protected $carsOnProperty = 0;

	public function checkIfCanEnter(EntityInterface $entity): bool {
		return $entity->canEnter();
	}

	public function letIn(EntityInterface $entity): void {
		$this->entitiesOnProperty[] = $entity;
		Log::info($entity->identify() . " let in.");

		if($entity instanceof Car) {
			$this->carsOnProperty++;
		}
	}

	public function countCars(): int {
		return $this->carsOnProperty;
	}
	
}

File: s3-inf-ppo/laboratoria/lab07/java/ppo7/src/ParkingLotManager/ParkingLot.java
package ParkingLotManager;

import ParkingLotManager.Entities.Car;
import ParkingLotManager.Interfaces.EntityInterface;

import java.util.ArrayList;

final public class ParkingLot {

    private ArrayList<EntityInterface> entitiesOnProperty = new ArrayList<>();
    private int carsOnProperty = 0;

    public boolean checkIfCanEnter(EntityInterface entity) {
        return entity.canEnter();
    }

    public void letIn(EntityInterface entity) {
        entitiesOnProperty.add(entity);
        Log.info(entity.identify() + " let in.");

        if(entity instanceof Car) {
            carsOnProperty++;
        }
    }

    public int countCars() {
        return carsOnProperty;
    }

}
